### PR TITLE
FAI-776 - Import explainability-core and explainability-integrationtests from kogito-apps

### PR DESCRIPTION
**Pull request by tteofili**
Importing explainability-core and explainability-integrationtests from _kogito-apps/explainability_.

### CODE REVIEW

**Comment by tteofili** [comment]
this is the first iteration of the work of migrating TrustyAI explainability library into its own repo.
Next steps would require setting up a proper parent, with dependency management, getting rid of kogito related dependencies where possible, setting up of different maven profiles (including one for releasing to Maven Central).

File: README.md [hunk 1]
```diff
@@ -2,11 +2,4 @@
 
 A java library for Explainable AI.
 
-It provides implementation of the following algorithms:
-
- * [LIME](https://arxiv.org/abs/1602.04938)
- * Constrainsts-based counterfactual search
- * [SHAP](https://arxiv.org/abs/1705.07874)
- * [PDP](https://www.jstor.org/stable/2699986)
```
Comment (danielezonca) [inline README.md:10]: I saw you moved the list of available algorithms in the readme of `explainability-core`, can you add a link here to that readme? Like "To get more information see ..."

File: trustyaiexplainability-core/src/test/java/org/kie/trustyai/explainability/local/lime/optim/LimeConfigEntityFactoryTest.java [hunk 2]
```diff
@@ -49,24 +50,24 @@ void testConversion() {
     @Test
     void testWeightingEntities() {
         List<? extends LimeConfigEntity> entities = LimeConfigEntityFactory.createWeightingEntities(new LimeConfig());
-        assertThat(entities).isNotNull().hasSize(1);
+        Assertions.assertThat(entities).isNotNull().hasSize(1);
```
Comment (danielezonca) [inline trustyaiexplainability-core/src/test/java/org/kie/trustyai/explainability/local/lime/optim/LimeConfigEntityFactoryTest.java]: This change should not be necessary, there is already an import static of `assertThat`

File: trustyaiexplainability-core/src/test/java/org/kie/trustyai/explainability/utils/DataUtilsTest.java [hunk 3]
```diff
@@ -237,7 +238,7 @@ void testDoublesToFeatures() {
         for (Feature f : features) {
             assertNotNull(f);
             assertNotNull(f.getName());
-            assertEquals(Type.NUMBER, f.getType());
+            Assertions.assertEquals(Type.NUMBER, f.getType());
```
Comment (danielezonca) [inline trustyaiexplainability-core/src/test/java/org/kie/trustyai/explainability/utils/DataUtilsTest.java]: This change should not be necessary, there is already an import static of `assertEquals`

File: trustyaiexplainability-integrationtests/pom.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-apps-build-parent</artifactId>
+    <version>1.22.1.Final</version>
+  </parent>
```
Comment (danielezonca) [inline trustyaiexplainability-integrationtests/pom.xml]: To be changed to target root pom

**Review by danielezonca — APPROVED** [review]
Minor comments but LGTM 👍

**Review by ruivieira — APPROVED** [review]
LGTM 👍, thanks @tteofili

**Pull request merged**